.ci/test_r_package.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+# set up R environment
+R_LIB_PATH=~/Rlib
+mkdir -p $R_LIB_PATH
+echo "R_LIBS=$R_LIB_PATH" > ${HOME}/.Renviron
+echo 'options(repos = "https://cran.rstudio.com")' > ${HOME}/.Rprofile
+export PATH="$R_LIB_PATH/R/bin:$PATH"
+
+# installing precompiled R for Ubuntu
+# https://cran.r-project.org/bin/linux/ubuntu/#installation
+# adding steps from https://stackoverflow.com/a/56378217/3986677 to get latest version
+if [[ $OS_NAME == "linux" ]]; then
+    sudo apt-get update
+    sudo add-apt-repository \
+        "deb https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/"
+    sudo apt-key adv \
+        --keyserver keyserver.ubuntu.com \
+        --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9
+    sudo apt-get update
+    sudo apt-get install \
+        -y \
```

Comment:
Add `--no-install-recommends`.

Reply:
added

Reply:
this is breaking things resulting in `Rscript: command not found`. I will try explicitly adding in `r-base-core` but if that doesn't work I think we should add it back in for now.

Reply:
and that is because I remove the `sudo installer` line accidentally! oy

Reply:
I see this error [on the linux side](https://travis-ci.org/microsoft/LightGBM/jobs/610688556?utm_medium=notification&utm_source=github_status):

```
W: GPG error: https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 51716619E084DAB9
E: The repository 'https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
Executing: /tmp/apt-key-gpghome.Lt8WVNv2MN/gpg.1.sh --keyserver keyserver.ubuntu.com --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9
gpg: key 51716619E084DAB9: public key "Michael Rutter <marutter@gmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Version '3.6.1-3bionic' for 'r-base-dev' was not found
```

I think it is the result of removing the second `update`, so I believe we need to leave it.

Reply:
> I think it is the result of removing the second update, so I believe we need to leave it.

Yeah! It's quite expected. The second `update` should be left and the first one should be removed.

Reply:
hmmmm ok let me try